packages/flutter_tools/lib/src/commands/build_setup_split_aot.dart

```diff
@@ -0,0 +1,136 @@
+        allowed: <String>['android-arm', 'android-arm64', 'android-x86', 'android-x64'],
+        help: 'The target platform for which the app is compiled.',
+      );
+    usesTrackWidgetCreation(verboseHelp: verboseHelp);
+  }
+
+  @override
+  final String name = 'setupsplitaot';
+
+  @override
+  Future<Set<DevelopmentArtifact>> get requiredArtifacts async => <DevelopmentArtifact>{
+    DevelopmentArtifact.androidGenSnapshot,
+  };
+
+  @override
+  final String description = 'Build an Android split AAB file from your app.\n\n'
+    "This command can build debug and release versions of your application. 'debug' builds support "
+    "debugging and a quick development cycle. 'release' builds don't support debugging and are "
+    'suitable for deploying to app stores.';
+
+  @override
+  Future<Map<CustomDimensions, String>> get usageValues async {
+    final Map<CustomDimensions, String> usage = <CustomDimensions, String>{};
+
+    usage[CustomDimensions.commandBuildApkTargetPlatform] =
+        stringsArg('target-platform').join(',');
+    usage[CustomDimensions.commandBuildApkSplitPerAbi] =
+        boolArg('split-per-abi').toString();
+
+    if (boolArg('release')) {
+      usage[CustomDimensions.commandBuildApkBuildMode] = 'release';
+    } else if (boolArg('debug')) {
+      usage[CustomDimensions.commandBuildApkBuildMode] = 'debug';
+    } else if (boolArg('profile')) {
+      usage[CustomDimensions.commandBuildApkBuildMode] = 'profile';
+    } else {
+      // The build defaults to release.
+      usage[CustomDimensions.commandBuildApkBuildMode] = 'release';
+    }
+    return usage;
+  }
+
+  @override
+  Future<FlutterCommandResult> runCommand() async {
+    if (globals.androidSdk == null) {
+      exitWithNoSdkMessage();
+    }
+    final BuildInfo buildInfo = getBuildInfo();
+    final AndroidBuildInfo androidBuildInfo = AndroidBuildInfo(
+      buildInfo,
+      splitPerAbi: boolArg('split-per-abi'),
+      targetArchs: stringsArg('target-platform').map<AndroidArch>(getAndroidArchForName),
+      shrink: false,
+      splitAot: true,
+    );
+    validateBuild(androidBuildInfo);
+
+
+    AssembleCommand assembleCommand = AssembleCommand();
+    FlutterCommandRunner runner = FlutterCommandRunner();
+    runner.addCommand(assembleCommand);
+    // 'android_aot_release_android-arm64'
+    FlutterProject project = FlutterProject.current();
+    Directory intermediateDir = project.directory.childDirectory('build').childDirectory('app').childDirectory('intermediates').childDirectory('flutter').childDirectory('release');
+    List<String> args = [
+      'assemble',
+      '--depfile=${intermediateDir.childFile('flutter_build.d').path}',
+      '--output=${intermediateDir.path}',
+      '-dTargetPlatform=android',
+      '-dBuildMode=release',
+      '--ExtraGenSnapshotOptions=--no-use-bare-instructions',
+      '--SplitAot=true',
+      '--SetupSplitAot=true',
+      'android_aot_bundle_release_android-arm'
```

Comment:
this is only a single arch

Reply:
Only a single arch is needed here as we just need to run gen_snapshot and grab the loading unit manifest output. Any arch build will produce the same loading unit manifest.